app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -148,7 +148,7 @@ register_katello_host(){
   params << "update_packages=#{@update_packages}" unless @update_packages.nil?
 -%>
   UUID=$(subscription-manager identity | grep --max-count 1 --only-matching '\([[:xdigit:]]\{8\}-[[:xdigit:]]\{4\}-[[:xdigit:]]\{4\}-[[:xdigit:]]\{4\}-[[:xdigit:]]\{12\}\)')
-  <%= indent(2, skip1: true) { generate_web_request(utility: @download_utility, url: @registration_url, ssl_ca_cert: "$KATELLO_SERVER_CA_CERT", headers: headers, params: params) } %>
+  <%= indent(2, skip1: true) { generate_web_request(utility: @download_utility, url: @registration_url, ssl_ca_cert: "$KATELLO_SERVER_CA_CERT", output_file: "/tmp/registration_host_init.sh", headers: headers, params: params) } %>
```

Comment:
Please don't write to pre-defined paths accessible by others.

A random user of the system can pre-create a `/tmp/registration_host_init.sh` (owned by said user), wait for the registration to start (which will write the script to the user-owned `/tmp/registration_host_init.sh`) and append own commands to it before the script is executed in line 166, thus gaining elevated privileges on the machine.

(well, at least in theory, I think modern Linux prevents many of those attacks, but I'd still avoid it)

Reply:
Since we require the user to be a `root`, the `/root/registration_host_init.sh` path should be fine, right?

Reply:
I think so, yes.

We use the same in app/views/unattended/provisioning_templates/snippet/ansible_provisioning_callback.erb:
```
<%= save_to_file('/root/ansible_provisioning_call.sh', snippet('ansible_tower_callback_script')) %>
```